app/javascript/controllers/map_controller.js

```diff
@@ -0,0 +1,96 @@
+import { Controller } from "stimulus"
+import consumer from "../channels/consumer"
+
+export default class extends Controller {
+  static targets = ["image"]
+
+  connect() {
+    this.mapId = this.element.dataset.mapId
+    this.channel = consumer.subscriptions.create({
+      channel: "MapChannel",
+      id: this.mapId
+    }, {
+      received: this.cableReceived.bind(this)
+    })
+  }
+
+  moveMap(event) {
+    const speedFactor = 2
+    var mousedown = {
+      x: event.pageX,
+      y: event.pageY
+    }
+    var backgroundPosition = {
+      x: Number.parseInt(event.target.style.backgroundPosition.split(" ")[0]),
+      y: Number.parseInt(event.target.style.backgroundPosition.split(" ")[1])
```

Comment:
How do you feel about keeping the coordinates as data attributes?

Reply:
Yes, this'll be better. Done in 44f6aa1. I didn't change zooming to use data attributes yet because I'm going to change the way zooming works.